Add a tcpdump verification FAQ entry for mutual TLS effect proof. (#6922)
istio/istio.io: content/en/faq/security/verify-mtls-encryption.md

---
title: How can I verify Istio mTLS indeed encrypt the traffic?
weight: 40
---

Once you turned on the mutual TLS, traffic is encrypted. We will show that through `tcpdump` in the
sidecar container.

To enable tcpdump, you need to install Istio with proxy priviledged enabled.

{{< text bash >}}
$ istioctl manifest apply --set values.global.proxy.privileged=true
{{< /text >}}

{{< warning >}}
This installation is only for demo purpose, please revert the change after you are done for better security!
{{< /warning>}}

> Deploy httpbin, sleep workloads, and a sleep-legacy deployment which does not have sidecar injected.

Reviewer: I would refer to the authentication task instead. IMO, the faq should be short, and skip demo set up. Otherwise, you can make this become a task by itself (we used to have mtls deepdive that show how to debug/verify mTLS, though it is deleted recently)

Reviewer 2: This also doesn't really sound like an answer to the general question: `How can I verify Istio mTLS indeed encrypt the traffic?` In a production env, users can't do this, so this is more of a simple demo task, not a general answer to a frequently asked question. Also, do we really need a task to demonstrate how to use tcpdump, or would a tip in the tls migration doc be good enough (for example, `you can verify that traffic is encrypted by running tcpdump in a proxy container, but only in an Istio installation with values.global.proxy.privileged=true`)?

Author: Some of our FAQ page does include the some cli command, though i agree this is a bit longer than others.

i'm fine to move the info to a task page. but would it be appropriate  to put a link in the FAQ page?

I just have seen many many users are not so sure about our mTLS effect. They feel, I know request can still pass, you take care of mtls, but how can i know learn more?

Author: moved to the migration doc.

$ kubectl apply -f ./samples/httpbin/httpbin.yaml
$ kubectl apply -f ./samples/sleep/sleep.yaml
$ kubectl apply -f - <<EOF
apiVersion: apps/v1
kind: Deployment
metadata:
  name: sleep-legacy
spec:
  replicas: 1
  selector:
    matchLabels:
      app: sleep-legacy
  template:
    metadata:
      annotations:
        sidecar.istio.io/inject: "false"
      labels:
        app: sleep-legacy
    spec:
      containers:
      - name: sleep
        image: governmentpaas/curl-ssl
        command: ["/bin/sleep", "3650d"]
        imagePullPolicy: IfNotPresent
EOF
serviceaccount/httpbin unchanged
service/httpbin created
deployment.apps/httpbin created
serviceaccount/sleep unchanged
service/sleep created
deployment.apps/sleep created
deployment.apps/sleep-legacy created
{{< /text >}}

Configuring peer authentication as `STRICT`.

{{< text bash >}}
$ kubectl apply -f - <<EOF
apiVersion: "security.istio.io/v1beta1"
kind: "PeerAuthentication"
metadata:
  name: "default"
spec:
  mtls:
    mode: STRICT
EOF
peerauthentication.security.istio.io/default created
{{< /text >}}

Now httpbin is protected under mutual TLS. We execute tcpdump in httpbin pod sidecar container.

{{< text bash >}}

> $ kubectl exec $(kubectl get pod  -lapp=sleep -ojsonpath={.items..metadata.name}) -c istio-proxy -it -- sudo tcpdump dst port 80  -A

Reviewer: Does this assume tcpdump is always on the proxy image?

Author: i think so. i'm using 1.5.0 build. distroless might be a bit different

---

> title: How can I verify Istio mTLS indeed encrypt the traffic?

Reviewer 2: ```suggestion
title: How can I verify that traffic is using mutual TLS encryption?
```

> weight: 40

Reviewer 2: ```suggestion
weight: 25
```

---

> See [Istio mutual TLS migration](/docs/tasks/security/authentication/mtls-migration), where we

Reviewer 2: ```suggestion
If you installed Istio with `values.global.proxy.privildeged=true`, you can use `tcpdump` to determine encryption status. See [Istio mutual TLS migration](/docs/tasks/security/authentication/mtls-migration) for instructions.
```

> provide instructions on how to use `tcpdump` to determine encryption status.

Reviewer 2: ```suggestion
```
